internal/cache/base.go

```diff
@@ -0,0 +1,206 @@
+package cache
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"fyne.io/fyne/v2"
+)
+
+var (
+	cacheDuration     = 1 * time.Minute
+	cleanTaskInterval = cacheDuration / 2
+
+	canvasRefreshCh = make(chan struct{}, 1)
+	expiredObjects  = make([]fyne.CanvasObject, 0, 50)
+	lastClean       time.Time
+)
+
+func init() {
+	if t, err := time.ParseDuration(os.Getenv("FYNE_CACHE")); err == nil {
+		cacheDuration = t
+		cleanTaskInterval = cacheDuration / 2
+	}
+}
+
+// CleanTask run cache clean task, it should be called on paint events.
+func CleanTask() {
+	now := time.Now()
+	// do not run clean task so fast
+	if now.Sub(lastClean) < 10*time.Second {
+		return
+	}
+	canvasRefresh := false
+	select {
+	case <-canvasRefreshCh:
+		canvasRefresh = true
+	default:
+		if now.Sub(lastClean) < cleanTaskInterval {
+			return
+		}
+	}
+	destroyExpiredRenderers(now)
+	destroyExpiredSvgs(now)
```

Comment:
Should expired renderers and svgs be removed with no synchronization with canvas.Refresh? Basically occurs two things:
1. In the way it is now, after some time (with no user interaction) widgets renderers will be removed so when user starts interaction again he would look like a blink because all the needed renderers will be created at that time.
2. If we remove expired renderers only when a canvas refresh is executed, then user would not see anything strange, but if no user interaction, the expired renderers will not be removed until a canvas.Refresh.

So both of them have disadvantages and I am not sure which choose. Any suggestions?

Reply:
Or we can delete renderers immediately when we detect they are not in the objects tree anymore, what do you think?

Reply:
Caching is a careful balance. We need to remove things that are not needed - and quickly - but keep things still used that may not have actually been refreshed recently.

If there is a way to do this without a timer that might be interesting, but if we need a timer then I don't think the problem above can be avoided.

Reply:
Hmm are you talking about option 1 or 2?

Reply:
Neither, I was commenting on how there is no right answer.
I guess 2 is a slightly better compromise as it is unlikely that the cache would grow without items being refreshed :).

Reply:
Agree :)

Reply:
done in 2a3abfb